core/src/main/java/com/tickaroo/tikxml/XmlReader.java

```diff
@@ -801,7 +801,7 @@ private int nextNonWhitespace(boolean throwOnEof, boolean isDocumentBeginning) t
       }
 
       buffer.skip(p - 1);
-      if (c == '<' && !isCDATA()) {
+      if (c == '<' && !isCDATA() && fillBuffer(2)) {
```

Comment:
Why number magic number 2?

Reply:
Good question. I wanted to ask the sam.
@sockeqwe you removed filledBuffer in the following commit
https://github.com/Tickaroo/tikxml/commit/c2565b2267797f7b3f639813d6a05045031eccea

i don´t know what exactly does this fillBuffer method so maybe you can have a short look